lib/AST/Module.cpp

```diff
@@ -804,6 +822,17 @@ void SourceFile::getTopLevelDecls(SmallVectorImpl<Decl*> &Results) const {
   Results.append(decls.begin(), decls.end());
 }
 
+void SourceFile::getDisplayDecls(SmallVectorImpl<Decl*> &Results) const {
+  if (Imports.hasValue()) {
+    for (auto import : *Imports) {
+      if (import.options.contains(ImportFlags::Exported)) {
```

Comment:
Will this potentially include the same decl more than once if it's imported via two different paths, eg

```
// A.swift:
@_exported import C

// B.swift
@_exported import C

// Foo.swift
import A
import B
```

Perhaps a better way is to get the 'flat' list of transitively-imported modules from the ImportCache, and then just get the top-level decls of each one?

Reply:
It looks like it will, based on how the `-import-underlying-module` case works (it currently crawls those modules for each file). I'll work on an alternative method to collect these modules ahead of time.

Reply:
Try something like this

```
auto &importCache = sf.getASTContext().getImportCache();
auto &importSet = importCache.getImportSet(sf);
auto modules = importSet.getTransitiveImports();
```

Reply:
Are the transitive imports what we want? I'm concerned about collecting decls that aren't actually part of the file/module, since they wouldn't be `@_exported`.

Reply:
The transitive imports are the @_exported ones only, I think.

Reply:
It looks like that includes much more than the `@_exported` imports; the framework from the test case seems to be including all of Foundation.

Reply:
I think i have a solution, but i'm not sure where/how to test it. The symbol graph deduplicated these symbols anyway, and i'm not sure where else uses `getDisplayDecls()` that we could slot a test into.

Reply:
What about something like this in `getDisplayDecls()`:

```
#ifndef NDEBUG
llvm::DenseSet<Decl *> visited;
for (auto *decl : decls) {
  auto inserted = visited.insert(decl).second;
  assert(inserted);
}
#endif

Reply:
Excellent, thanks. The PR looks good to me.